homeassistant/components/subaru/config_flow.py

```diff
@@ -0,0 +1,179 @@
+        websession = aiohttp_client.async_get_clientsession(self.hass)
+        now = datetime.now()
+        if not data.get(CONF_DEVICE_ID):
+            data[CONF_DEVICE_ID] = int(now.timestamp())
+        date = now.strftime("%Y-%m-%d")
+        device_name = "Home Assistant: Added " + date
+
+        self.controller = SubaruAPI(
+            websession,
+            username=data[CONF_USERNAME],
+            password=data[CONF_PASSWORD],
+            device_id=data[CONF_DEVICE_ID],
+            pin=None,
+            device_name=device_name,
+            country=data[CONF_COUNTRY],
+        )
+        _LOGGER.debug("Using subarulink %s", self.controller.version)
+        _LOGGER.debug(
+            "Setting up first time connection to Subuaru API.  This may take up to 20 seconds."
+        )
+        if await self.controller.connect():
+            _LOGGER.debug("Successfully authenticated and authorized with Subaru API")
+            self.config_data.update(data)
+
+    async def async_step_pin(self, user_input=None):
+        """Handle second part of config flow, if required."""
+        error = None
+        if user_input:
+            if self.controller.update_saved_pin(user_input[CONF_PIN]):
+                try:
+                    vol.Match(r"[0-9]{4}")(user_input[CONF_PIN])
+                    await self.controller.test_pin()
+                except vol.Invalid:
+                    error = {"base": "bad_pin_format"}
+                except InvalidPIN:
+                    error = {"base": "incorrect_pin"}
+                else:
+                    _LOGGER.debug("PIN successfully tested")
+                    self.config_data.update(user_input)
+                    return self.async_create_entry(
+                        title=self.config_data[CONF_USERNAME], data=self.config_data
+                    )
+        return self.async_show_form(step_id="pin", data_schema=PIN_SCHEMA, errors=error)
+
+
+class OptionsFlowHandler(config_entries.OptionsFlow):
+    """Handle a option flow for Subaru."""
+
+    def __init__(self, config_entry: config_entries.ConfigEntry):
+        """Initialize options flow."""
+        self.config_entry = config_entry
+
+    async def async_step_init(self, user_input=None):
+        """Handle options flow."""
+        if user_input is not None:
+            return self.async_create_entry(title="", data=user_input)
+
+        data_schema = vol.Schema(
+            {
+                vol.Required(
+                    CONF_SCAN_INTERVAL,
+                    default=self.config_entry.options.get(
+                        CONF_SCAN_INTERVAL, DEFAULT_SCAN_INTERVAL
+                    ),
+                ): vol.All(cv.positive_int, vol.Clamp(min=MIN_SCAN_INTERVAL)),
+                vol.Required(
+                    CONF_HARD_POLL_INTERVAL,
+                    default=self.config_entry.options.get(
+                        CONF_HARD_POLL_INTERVAL, DEFAULT_HARD_POLL_INTERVAL
+                    ),
```

Comment:
As developers we should choose a reasonable default and not expose to the control unless its strictly needed as users tend to set it too low and get integrations banned from cloud services.

Reply:
I'm glad you brought this up, because these options were bothering me when I typed up the documentation for them. I've set the API polling to a fixed 5 minute interval, and now expose a boolean enable/disable (default disable) option to poll the vehicle with a remote command on a fixed 120 minute interval.